### ISSUE TITLE
Accept external EE library (user supplied)

### ISSUE THREAD

**Issue by ibgreen** [UnfoldedInc/earthengine-layers#45]
To support use in bigger applications that already use the ee API, we need to have ee library as a peerDependency and ask the user to pass it in to the layer as a prop.

**Comment by kylebarron** [comment]
> ask the user to pass it in to the layer as a prop

Why? It's not enough just to set `@google/earthengine` as a `peerDependency`? `@deck.gl/*` are currently set as `peerDependency`s, but we still are able to import them with `import ... from '@deck.gl/core'`.

So I would think we could do the same with EE.

**Comment by ibgreen** [comment]
> Why? It's not enough just to set @google/earthengine as a peerDependency

Yes it should absolutely be a peerDependency. Let's start with that and confirm with the reporting user that this solves the problem.

**Comment by ibgreen** [comment]
It may be that the user doesn't use npm but installs the library using a script mechanism or uses a debug version, and they may not have an advanced webpack setup to control that.

**Comment by kylebarron** [comment]
It would also be possible to declare `ee` as a global in the rollup config
https://github.com/UnfoldedInc/earthengine-layers/blob/a0f7d4f57a3642fcbdf49e6b5f11a9233e95575d/modules/earthengine-layers/rollup.config.js#L11-L16

I think we still need to include `ee` for pydeck, but one of the rollup builds could assume `ee` being global.

**Comment by kylebarron** [comment]
What should the name be for a bundle that omits `@google/earthengine`? `bundle_core.js` and `bundle_core.min.js`?

**Comment by ibgreen** [comment]
> What should the name be for a bundle that omits @google/earthengine?

For simplicity's and consistency's sake, let me make a soft push against having multiple versions of the script. 

Is there a recommended or obvious way to load EE from a script? If so can we just create one script that always requires the external script to be loaded, and just document this.

Generally it is preferable to not bundle such a big dependency in our script.

**Comment by kylebarron** [comment]
I couldn't find any documentation on loading EarthEngine from a script, but based on their [build config](https://github.com/google/earthengine-api/blob/5e15afc289661d8c6166069e4a3d6a3bb199993a/javascript/gruntfile.js#L57-L61), a working CDN url that presumably works in the browser is:
```
https://unpkg.com/@google/earthengine/build/browser.js
```

Publishing only one bundle, without including EarthEngine, would require a change to Pydeck's custom layers to accept an array of custom libraries, something like:
```js
{
	"libraryName": "EarthEngineLayerLibrary",
	"libraryUrls": ["https://unpkg.com/@google/earthengine/build/browser.js", "url to earthengine-layers bundle"]
} 
```
I believe that currently the custom layer accepts only a single string.

**Comment by kylebarron** [comment]
Fixed with #51
